tests/integration/wasm_tests.cpp

```diff
@@ -0,0 +1,355 @@
+#include "wasm_tests.h"
+
+#include <fcntl.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+
+#include <cmath>
+#include <cstdlib>
+#include <cstring>
+
+#include "../../debug.h"
+#include "../../instructions.h"
+#include "assertion.h"
+
+int COUNT = 0;
+
+uint8_t *mmap_file(char *path, int *len) {
+    int fd;
+    int res;
+    struct stat sb {};
+    uint8_t *bytes;
+
+    fd = open(path, O_RDONLY);
+    if (fd < 0) {
+        FATAL("could not open file '%s'\n", path);
+    }
+    res = fstat(fd, &sb);
+    if (res < 0) {
+        FATAL("could not stat file '%s' (%d)\n", path, res);
+    }
+
+    bytes = (uint8_t *)mmap(nullptr, sb.st_size, PROT_READ, MAP_SHARED, fd, 0);
+    if (len) {
+        *len = sb.st_size;  // Return length if requested
+    }
+    if (bytes == MAP_FAILED) {
+        FATAL("could not mmap file '%s'", path);
+    }
+    return bytes;
+}
+
+// Parse and add arguments to the stack
+void parse_args(Module *m, Type *type, int argc, Value argv[]) {
+    for (int i = 0; i < argc; i++) {
+        m->sp++;
+        StackValue *sv = &m->stack[m->sp];
+        sv->value_type = type->params[i];
+        switch (type->params[i]) {
+            case I64:
+                sv->value.int64 = argv[i].int64;
+                break;
+            case I32:
+                sv->value.int32 = argv[i].int32;
+                break;
+            case F32:
+                sv->value.f32 = argv[i].f32;
+                break;
+            case F64:
+                sv->value.f64 = argv[i].f64;
+                break;
+        }
+    }
+}
+
+void invoke(Module *m, const char *call_f, Value values[]) {
+    uint32_t fidx = -1;
+    m->sp = -1;
+    m->fp = -1;
+    m->csp = -1;
+    // TODO move to the WARDuino class

```

Comment:
This would indeed be a convenient addition to the WARDuino class

Reply:
When we move this, it makes sense to immediately add support for multiple modules as well. Adding this to issue #27.